langchain4j/src/main/java/dev/langchain4j/service/tool/ToolProviderResult.java

```diff
@@ -0,0 +1,44 @@
+package dev.langchain4j.service.tool;
+
+import dev.langchain4j.agent.tool.ToolSpecification;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A wrapper mostly used by the {@link ToolProvider}
+ */
+public class ToolProviderResult {
+    private final Map<ToolSpecification, ToolExecutor> tools;
+
+    public ToolProviderResult(Map<ToolSpecification, ToolExecutor> tools) {
+        this.tools = tools;
+    }
+
+    public Map<ToolSpecification, ToolExecutor> tools() {
+        return Collections.unmodifiableMap(tools);
+    }
+
+    public static ToolProviderResultBuilder builder() {
+        return new ToolProviderResultBuilder();
+    }
+
+    public static class ToolProviderResultBuilder {
+        private final Map<ToolSpecification, ToolExecutor> tools = new HashMap<>();
```

Comment:
```suggestion
        private final Map<ToolSpecification, ToolExecutor> tools = new LinkedHashMap<>();
```
Better to preserve the ordering defined by the user